<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Constructs PANDA gene regulatory networks from single-cell gene expression data — scorpion • SCORPION</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Constructs PANDA gene regulatory networks from single-cell gene expression data — scorpion"><meta name="description" content="Constructs gene regulatory networks from single-cell gene expression data using the PANDA (Passing Attributes between Networks for Data Assimilation) algorithm."><meta property="og:description" content="Constructs gene regulatory networks from single-cell gene expression data using the PANDA (Passing Attributes between Networks for Data Assimilation) algorithm."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SCORPION</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/quick-start.html">Get Started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/quick-start.html">Quick Start Guide</a></li>
    <li><a class="dropdown-item" href="../articles/algorithm.html">Algorithm &amp; Mathematics</a></li>
    <li><a class="dropdown-item" href="../articles/advanced-usage.html">Advanced Usage</a></li>
    <li><a class="dropdown-item" href="../articles/network-visualization.html">Network Visualization</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Zaoqu-Liu/SCORPION" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Constructs PANDA gene regulatory networks from single-cell gene expression data</h1>
      <small class="dont-index">Source: <a href="https://github.com/Zaoqu-Liu/SCORPION/blob/HEAD/R/scorpion.R" class="external-link"><code>R/scorpion.R</code></a></small>
      <div class="d-none name"><code>scorpion.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Constructs gene regulatory networks from single-cell gene expression data using the PANDA (Passing Attributes between Networks for Data Assimilation) algorithm.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">scorpion</span><span class="op">(</span></span>
<span>  tfMotifs <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">gexMatrix</span>,</span>
<span>  ppiNet <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  computingEngine <span class="op">=</span> <span class="st">"cpu"</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  gammaValue <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  nPC <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  assocMethod <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  alphaValue <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  hammingValue <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  nIter <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  outNet <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"regulatory"</span>, <span class="st">"coregulatory"</span>, <span class="st">"cooperative"</span><span class="op">)</span>,</span>
<span>  zScaling <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  showProgress <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  randomizationMethod <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>  scaleByPresent <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  filterExpr <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tfmotifs">tfMotifs<a class="anchor" aria-label="anchor" href="#arg-tfmotifs"></a></dt>
<dd><p>A motif dataset, a data.frame or a matrix containing 3 columns. Each row describes an motif associated with a transcription factor (column 1) a gene (column 2) and a score (column 3) for the motif.</p></dd>


<dt id="arg-gexmatrix">gexMatrix<a class="anchor" aria-label="anchor" href="#arg-gexmatrix"></a></dt>
<dd><p>An expression dataset, with genes in the rows and barcodes (cells) in the columns.</p></dd>


<dt id="arg-ppinet">ppiNet<a class="anchor" aria-label="anchor" href="#arg-ppinet"></a></dt>
<dd><p>A Protein-Protein-Interaction dataset, a data.frame or matrix containing 3 columns. Each row describes a protein-protein interaction between transcription factor 1(column 1), transcription factor 2 (column 2) and a score (column 3) for the interaction.</p></dd>


<dt id="arg-computingengine">computingEngine<a class="anchor" aria-label="anchor" href="#arg-computingengine"></a></dt>
<dd><p>Either 'cpu' or 'gpu'</p></dd>


<dt id="arg-ncores">nCores<a class="anchor" aria-label="anchor" href="#arg-ncores"></a></dt>
<dd><p>Number of processors to be used if BLAS or MPI is active.</p></dd>


<dt id="arg-gammavalue">gammaValue<a class="anchor" aria-label="anchor" href="#arg-gammavalue"></a></dt>
<dd><p>Graining level of data (proportion of number of single cells in the initial dataset to the number of super-cells in the final dataset)</p></dd>


<dt id="arg-npc">nPC<a class="anchor" aria-label="anchor" href="#arg-npc"></a></dt>
<dd><p>Number of principal components to use for construction of single-cell kNN network.</p></dd>


<dt id="arg-assocmethod">assocMethod<a class="anchor" aria-label="anchor" href="#arg-assocmethod"></a></dt>
<dd><p>Association method. Must be one of 'pearson', 'spearman' or 'pcNet'.</p></dd>


<dt id="arg-alphavalue">alphaValue<a class="anchor" aria-label="anchor" href="#arg-alphavalue"></a></dt>
<dd><p>Value to be used for update variable.</p></dd>


<dt id="arg-hammingvalue">hammingValue<a class="anchor" aria-label="anchor" href="#arg-hammingvalue"></a></dt>
<dd><p>Value at which to terminate the process based on Hamming distance.</p></dd>


<dt id="arg-niter">nIter<a class="anchor" aria-label="anchor" href="#arg-niter"></a></dt>
<dd><p>Sets the maximum number of iterations PANDA can run before exiting.</p></dd>


<dt id="arg-outnet">outNet<a class="anchor" aria-label="anchor" href="#arg-outnet"></a></dt>
<dd><p>A vector containing which networks to return. Options include "regulatory", "coregulatory", "cooperative".</p></dd>


<dt id="arg-zscaling">zScaling<a class="anchor" aria-label="anchor" href="#arg-zscaling"></a></dt>
<dd><p>Boolean to indicate use of Z-Scores in output. False will use [0,1] scale.</p></dd>


<dt id="arg-showprogress">showProgress<a class="anchor" aria-label="anchor" href="#arg-showprogress"></a></dt>
<dd><p>Boolean to indicate printing of output for algorithm progress.</p></dd>


<dt id="arg-randomizationmethod">randomizationMethod<a class="anchor" aria-label="anchor" href="#arg-randomizationmethod"></a></dt>
<dd><p>Method by which to randomize gene expression matrix. Default "None". Must be one of "None", "within.gene", "by.genes". "within.gene" randomization scrambles each row of the gene expression matrix, "by.gene" scrambles gene labels.</p></dd>


<dt id="arg-scalebypresent">scaleByPresent<a class="anchor" aria-label="anchor" href="#arg-scalebypresent"></a></dt>
<dd><p>Boolean to indicate scaling of correlations by percentage of positive samples.</p></dd>


<dt id="arg-filterexpr">filterExpr<a class="anchor" aria-label="anchor" href="#arg-filterexpr"></a></dt>
<dd><p>Boolean to indicate wheter or not to remove genes with 0 expression across all cells from the GEX input.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of matrices describing networks achieved by convergence with PANDA algorithm.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Daniel Osorio &lt;daniecos@uio.no&gt;</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Loading example data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">scorpionTest</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># The structure of the data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">scorpionTest</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ gex:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ i       : int [1:4456] 1 5 8 11 22 30 33 34 36 38 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ p       : int [1:81] 0 47 99 149 205 258 306 342 387 423 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ Dim     : int [1:2] 230 80</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ Dimnames:List of 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. .. ..$ : chr [1:230] "MS4A1" "CD79B" "CD79A" "HLA-DRA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. .. ..$ : chr [1:80] "ATGCCAGAACGACT" "CATGGCCTGTGCAT" "GAACCTGATGAACC" "TGACTGGATTCTCA" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ x       : num [1:4456] 1 1 3 1 1 4 1 5 1 1 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..@ factors : list()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ tf :'data.frame':	4485 obs. of  3 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ tf    : chr [1:4485] "ADNP" "ADNP" "ADNP" "AEBP2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ target: chr [1:4485] "PRF1" "TMEM40" "TNFRSF1B" "CFP" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ mor   : num [1:4485] 1 1 1 1 1 1 1 1 1 1 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ ppi:'data.frame':	12754 obs. of  3 variables:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ protein1      : chr [1:12754] "ADNP" "ADNP" "ADNP" "AEBP2" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ protein2      : chr [1:12754] "ZBTB14" "NFIA" "CDC5L" "YY1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..$ combined_score: num [1:12754] 0.769 0.64 0.581 0.597 0.54 0.753 0.659 0.548 0.59 0.654 ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># List of 3</span></span></span>
<span class="r-in"><span><span class="co"># $ gex:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ i       : int [1:4456] 1 5 8 11 22 30 33 34 36 38 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ p       : int [1:81] 0 47 99 149 205 258 306 342 387 423 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dim     : int [1:2] 230 80</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dimnames:List of 2</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:230] "MS4A1" "CD79B" "CD79A" "HLA-DRA" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:80] "ATGCCAGAACGACT" "CATGGCCTGTGCAT" "GAACCTGATGAACC" "TGACTGGATTCTCA" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ x       : num [1:4456] 1 1 3 1 1 4 1 5 1 1 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ factors : list()</span></span></span>
<span class="r-in"><span><span class="co"># $ tf :'data.frame':  4485 obs. of  3 variables:</span></span></span>
<span class="r-in"><span><span class="co">#   ..$ tf    : chr [1:4485] "ADNP" "ADNP" "ADNP" "AEBP2" ...</span></span></span>
<span class="r-in"><span><span class="co"># ..$ target: chr [1:4485] "PRF1" "TMEM40" "TNFRSF1B" "CFP" ...</span></span></span>
<span class="r-in"><span><span class="co"># ..$ mor   : num [1:4485] 1 1 1 1 1 1 1 1 1 1 ...</span></span></span>
<span class="r-in"><span><span class="co"># $ ppi:'data.frame':  12754 obs. of  3 variables:</span></span></span>
<span class="r-in"><span><span class="co">#   ..$ X.node1       : chr [1:12754] "ADNP" "ADNP" "ADNP" "AEBP2" ...</span></span></span>
<span class="r-in"><span><span class="co"># ..$ node2         : chr [1:12754] "ZBTB14" "NFIA" "CDC5L" "YY1" ...</span></span></span>
<span class="r-in"><span><span class="co"># ..$ combined_score: num [1:12754] 0.769 0.64 0.581 0.597 0.54 0.753 0.659 0.548 0.59 0.654 ...</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Running SCORPION with large alphaValue for testing purposes.</span></span></span>
<span class="r-in"><span><span class="va">scorpionOutput</span> <span class="op">&lt;-</span> <span class="fu">scorpion</span><span class="op">(</span>tfMotifs <span class="op">=</span> <span class="va">scorpionTest</span><span class="op">$</span><span class="va">tf</span>,</span></span>
<span class="r-in"><span>                           gexMatrix <span class="op">=</span> <span class="va">scorpionTest</span><span class="op">$</span><span class="va">gex</span>,</span></span>
<span class="r-in"><span>                           ppiNet <span class="op">=</span> <span class="va">scorpionTest</span><span class="op">$</span><span class="va">ppi</span>,</span></span>
<span class="r-in"><span>                           alphaValue <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">--</span> <span style="font-weight: bold;">SCORPION</span> <span style="color: #00BBBB;">--------------------------------------------------------------------</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> Initializing and validating</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> Verified sufficient samples</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> Normalizing networks</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> Learning Network</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> Using tanimoto similarity</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 0: hamming distance = 0.73486</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 1: hamming distance = 0.22155</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 2: hamming distance = 0.078</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 3: hamming distance = 0.01924</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 4: hamming distance = 0.00414</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 5: hamming distance = 0.00085</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">v</span> Successfully ran SCORPION on 214 Genes and 783 TFs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 5: hamming distance = 0.00085
</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">i</span> Time elapsed: 2.5 seconds</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Iteration 5: hamming distance = 0.00085
</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># -- SCORPION --------------------------------------------------------------------------------------</span></span></span>
<span class="r-in"><span><span class="co"># + Initializing and validating</span></span></span>
<span class="r-in"><span><span class="co"># + Verified sufficient samples</span></span></span>
<span class="r-in"><span><span class="co"># i Normalizing networks</span></span></span>
<span class="r-in"><span><span class="co"># i Learning Network</span></span></span>
<span class="r-in"><span><span class="co"># i Using tanimoto similarity</span></span></span>
<span class="r-in"><span><span class="co"># + Successfully ran SCORPION on 214 Genes and 783 TFs</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Structure of the output.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">scorpionOutput</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ regNet  : num [1:783, 1:214] -0.413 1.517 -1.311 0.364 -1.041 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..- attr(*, "dimnames")=List of 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ coregNet: num [1:214, 1:214] 7.07e+06 -4.06 1.76e+01 -1.16e+01 -1.62e+01 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..- attr(*, "dimnames")=List of 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ coopNet : num [1:783, 1:783] 5.65e+06 -5.16 -3.79 -3.63 2.94 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ..- attr(*, "dimnames")=List of 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ numGenes: int 214</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ numTFs  : int 783</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  $ numEdges: int 167562</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># List of 6</span></span></span>
<span class="r-in"><span><span class="co"># $ regNet  :Formal class 'dgeMatrix' [package "Matrix"] with 4 slots</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ x       : num [1:167562] -0.413 1.517 -1.311 0.364 -1.041 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dim     : int [1:2] 783 214</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dimnames:List of 2</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ factors : list()</span></span></span>
<span class="r-in"><span><span class="co"># $ coregNet:Formal class 'dgeMatrix' [package "Matrix"] with 4 slots</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ x       : num [1:45796] 7.07e+06 -4.06 1.76e+01 -1.16e+01 -1.62e+01 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dim     : int [1:2] 214 214</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dimnames:List of 2</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:214] "ACAP1" "ACRBP" "ACSM3" "ADAR" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ factors : list()</span></span></span>
<span class="r-in"><span><span class="co"># $ coopNet :Formal class 'dgeMatrix' [package "Matrix"] with 4 slots</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ x       : num [1:613089] 5.65e+06 -5.16 -3.79 -3.63 2.94 ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dim     : int [1:2] 783 783</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ Dimnames:List of 2</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. .. ..$ : chr [1:783] "ADNP" "AEBP2" "AIRE" "ALX1" ...</span></span></span>
<span class="r-in"><span><span class="co"># .. ..@ factors : list()</span></span></span>
<span class="r-in"><span><span class="co"># $ numGenes: int 214</span></span></span>
<span class="r-in"><span><span class="co"># $ numTFs  : int 783</span></span></span>
<span class="r-in"><span><span class="co"># $ numEdges: int 167562</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/Zaoqu-Liu" class="external-link">Zaoqu Liu</a></p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

